jax/_src/numpy/linalg.py

```diff
@@ -23,43 +23,28 @@
 from jax import jit, custom_jvp
 from jax import lax
 
-from jax._src import dtypes
 from jax._src.lax import lax as lax_internal
 from jax._src.lax import linalg as lax_linalg
 from jax._src.numpy import lax_numpy as jnp
-from jax._src.numpy.util import _wraps
+from jax._src.numpy.util import _wraps, _promote_dtypes_inexact
 from jax._src.util import canonicalize_axis
 
 _T = lambda x: jnp.swapaxes(x, -1, -2)
 _H = lambda x: jnp.conjugate(jnp.swapaxes(x, -1, -2))
 
 
-def _promote_arg_dtypes(*args):
-  """Promotes `args` to a common inexact type."""
-  dtype, weak_type = dtypes._lattice_result_type(*args)
-  if not jnp.issubdtype(dtype, jnp.inexact):
-    dtype, weak_type = jnp.float_, False
-  dtype = dtypes.canonicalize_dtype(dtype)
-  args = [lax_internal._convert_element_type(arg, dtype, weak_type)
-          for arg in args]
-  if len(args) == 1:
-    return args[0]
-  else:
-    return args
-
-
 @_wraps(np.linalg.cholesky)
 @jit
 def cholesky(a):
-  a = _promote_arg_dtypes(jnp.asarray(a))
+  a = _promote_dtypes_inexact(jnp.asarray(a))[0]
```

Comment:
I have a slight preference here for writing `a, = _promote_dtypes_inexact(jnp.asarray(a))` because it includes an implicit assertion about the length of the returned tuple.

Reply:
Ah, I have seen that JAX use `a, =` in other file, but I have known the reason before. I use `...[0]` because `a, =` make `=` not align. But consider the reason your provide I think I should use `a, =`